samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -105,12 +105,12 @@ public void tryBecomeLeader(LeaderElectorListener leaderElectorListener) {
     LOG.info("Index = " + index + " Not eligible to be a leader yet!");
     String predecessor = children.get(index - 1);
     if (!predecessor.equals(currentSubscription)) {
-      if (currentSubscription != null) {
-
-        // callback in case if the previous node gets deleted (when previous processor dies)
-        if (previousProcessorChangeListener == null)
-          previousProcessorChangeListener =  new PreviousProcessorChangeListener(leaderElectorListener);
+      
+      // callback in case if the previous node gets deleted (when previous processor dies)
+      if (previousProcessorChangeListener == null)
+        previousProcessorChangeListener =  new PreviousProcessorChangeListener(leaderElectorListener);
```

Comment:
Good observation. I reviewed the code - and it looks like the original logic is wrong. This was just to fix the NPE.
So I changed it to be more correct for existing logic.
If your change coming in today - sure, let's just wait for it. If it is not yet, let's review and commit this change, because it needed in testing (to avoid NPEs).